firebase/src/core/domain/models/memo.ts

```diff
@@ -1,16 +1,36 @@
 import { defaultMaxStringLength, validate } from "#utils/validate";
 import * as Joi from "joi";
 
+interface MemoContent {
+  insert: string;
+  attributes?: {
+    bold?: boolean;
+    italic?: boolean;
+    underline?: boolean;
+    codeBlock?: boolean;
+  };
+}
+
 export interface Memo {
-  readonly id: string;
-  readonly question: Record<string, unknown>[];
-  readonly answer: Record<string, unknown>[];
+  readonly uniqueId: string;
+  readonly question: MemoContent[];
+  readonly answer: MemoContent[];
 }
 
+const memoContentValidationSchema = Joi.object({
```

Comment:
```suggestion
const memoQuillValidationSchema = Joi.object({
```

Just to enforce where we are taking these from. I think being specific here is beneficial.

Maybe also add a succinct doc?

Reply:
> Just to enforce where we are taking these from. I think being specific here is beneficial.

But aren't we going to follow the same format when we develop our own editor in https://github.com/olmps/memo/issues/223? If yes, I don't see why would we enforce `quill` in the name here, since we plan to remove it in the near future

Reply:
Renamed